### PR TITLE
117 slam refactoring

### PR DESCRIPTION
**Pull request by katethemate**

### CODE REVIEW

File: SLAM/position_evaluator.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    alpha_B = calculate_alpha(landmark.B.rotation)
+    ZBx = math.sin(alpha_B)*landmark.B.distance
+    ZBy = math.cos(alpha_B)*landmark.B.distance
+    robot_B = [landmark.B.x+ZBx,landmark.B.y+ZBy]
+
+    alpha_C = calculate_alpha(landmark.C.rotation)
+    ZCx = math.sin(alpha_C) * landmark.C.distance
+    ZCy = math.cos(alpha_C) * landmark.C.distance
+    robot_C = [landmark.B.x+ZCx,landmark.C.y+ZCy]
+
```
Comment (elli-wolf) [inline SLAM/position_evaluator.py:20]: This function doesn't return the robot position.

File: SLAM/landmark_matcher.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+def match_landmarks(landmarks, stored_landmarks):
+    #compare vectors
+    for landmark in landmarks:
+        for existing_landmark in stored_landmarks:
+            if landmark.AB==existing_landmark.AB and landmark.BC==landmark.BC:
+                SLAM.position_evaluator(landmark)
+            else:
+                total_landmarks.append(landmark)
+    return total_landmarks
```
Comment (elli-wolf) [inline SLAM/landmark_matcher.py]: If this function calls the postion_evaluator, but only returns the total_landmarks - how can we forward the new robot position to the pointcloud_generator or sender?

**Review by elli-wolf — CHANGES_REQUESTED** [review]

File: SLAM/position_evaluator.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    ZBy = math.cos(alpha_B)*landmark.B.distance
+    robot_B = Point(None, None, x=landmark.B.x+ZBx, y=landmark.B.y+ZBy)
+
+    alpha_C = calculate_alpha(landmark.C.rotation)
+    ZCx = math.sin(alpha_C) * landmark.C.distance
+    ZCy = math.cos(alpha_C) * landmark.C.distance
+    robot_C = Point(None, None, x=landmark.B.x+ZCx, y=landmark.C.y+ZCy)
+
+    robot_position = (robot_A+robot_B+robot_C)
+    return robot_position
```
Comment (elli-wolf) [inline SLAM/position_evaluator.py]: You are currently returning the sum of the three estimated positions (`robot_A`, `robot_B`, `robot_C`), but I think you actually want to return the average of the three positions.

**Review by elli-wolf — CHANGES_REQUESTED** [review]

File: SLAM/position_evaluator.py (hunk 3 above)
Comment (katethemate) [reply SLAM/position_evaluator.py]: oh yes of course thanks :)

**Review by elli-wolf — APPROVED** [review]
Looks good to me 👍

**Pull request merged**